macos/FluentUI/Button/Button.swift

```diff
@@ -139,6 +146,32 @@ open class Button: NSButton {
 			}
 		}
 	}
+	
+	/// the current state of the button
+	private var buttonState: ButtonState? {
+		didSet {
+			guard oldValue != buttonState else {
+				return
+			}
+			updateImageForButtonState()
+		}
+	}
+	
+	/// Dict of images for different button states
+	private var imagesForButtonState = [ButtonState: NSImage]()
+	
+	public func addImage(forButtonState: ButtonState, image: NSImage) {
```

Comment:
Could this be encapsulated in one public variable of type [ButtonState: NSImage] with a getter and setter?